gtime/model_selection/tests/test_cross_validation.py

```diff
@@ -0,0 +1,80 @@
+import pandas as pd 
+import numpy as np
+import pytest 
+from gtime.model_selection import time_series_split
+
+
+class TestTimeSeriesSplit:
+    def _correct_split_on_time_record_length(self):
+        date_rng = pd.date_range(start='1/1/2018', end='1/08/2018', freq='D')
+        time_series = pd.DataFrame(date_rng, columns=['date'])
+        time_series.set_index('date', inplace=True)
+        time_series['data'] = np.random.randint(0,100,size=(len(date_rng)))
+        n_splits = 5
+        start_date = time_series.index[0] 
+        end_date = time_series.index[-1]
+        split_length = ((end_date - start_date) / n_splits)
+        next_date = start_date + pd.Timedelta(split_length)
+
+        for split in range(n_splits - 1):
+            time_fold = time_series[(time_series.index >= start_date) & (time_series.index < next_date)]
+            yield time_fold.index
+            next_date += pd.Timedelta(split_length)
+        last_time_fold = time_series[0:]
+        yield last_time_fold.index
+
+    def test_samples_split_counts(self):
+        date_rng = pd.date_range(start='1/1/2018', end='1/08/2018', freq='D')
+        time_series = pd.DataFrame(date_rng, columns=['date'])
+        time_series.set_index('date', inplace=True)
+        time_series['data'] = np.random.randint(0,100,size=(len(date_rng)))
+
+        with pytest.raises(ValueError):
+            for element in (time_series_split(time_series, n_splits=(len(time_series)+1), split_on='index')):
+                pass
+
+    def check_split_on_time_index_type(self):
+        # If the split_on is set to 'time' but index is not DateTime
+        date_rng = pd.date_range(start='1/1/2018', end='1/08/2018', freq='D')
+        time_series = pd.DataFrame(date_rng, columns=['date'])
+        time_series['data'] = np.random.randint(0,100,size=(len(date_rng)))
+
+        with pytest.raises(ValueError):
+            for element in (time_series_split(time_series, n_splits=5, split_on='time')):
+                pass
+
+    def split_on_index(self):
+        date_rng = pd.date_range(start='1/1/2018', end='1/08/2018', freq='D')
```

Comment:
All this initialization part has to go in a `fixture` or you have to use hypothesis

Reply:
Done